Updated FAQ Client section

foss- opened the pull request:
- improved intro text
- removed Metatext iOS client (never worked without errors for friendica)
- added Mastodon iOS client
- corrected link for iOS Stella
- updated Tooot link, added data collection hint
- Tootle: removed hint and added update year
- corrected Stella* name
- removed StarPterano (app not updated for a while, no privacy policy provided to apple, UI lacking, login on every open required - not a user experience to recommend)
- Sailfish Friendly: improved link, added last update year
- macOS added Whalebird
- updated german faq

foss- committed:
Updated FAQ Client section

- improved intro text
- removed Metatext iOS client (never worked without errors for friendica)
- added Mastodon iOS client
- corrected link for iOS Stella
- updated Tooot link, added data collection hint
- Tootle: removed hint and added update year
- corrected Stella* name
- removed StarPterano (app not updated for a while, no privacy policy provided to apple, UI lacking, login on every open required - not a user experience to recommend)
- Sailfish Friendly: improved link, added last update year
- macOS added Whalebird
- updated german faq

foss- committed:
Merge branch 'friendica:develop' into doc_faq_client-clean-up

MrPetovan reviewed (commented):
What is the rationale behind forcing all app links to the store? I'd prefer if we kept the link to their official homepage.

foss- committed:
FAQ Client list unified format

- iOS and macOS listings now follow the adnroid listing format
- missing information added

tobiasd commented on doc/FAQ.md:
```diff
@@ -171,9 +171,7 @@ Example: Friendica Support
-Friendica is using a [Twitter/GNU Social compatible API](help/api), which means you can use any Twitter/GNU Social client for your platform as long as you can change the API path in its settings.
-Since the 2021.06 release, Friendica also supports the [Mastodon API](help/API-Mastodon) which works with many Mastodon clients.
-Here is a list of known working clients:
+Friendica supports [Mastodon API](help/API-Mastodon) and [Twitter API](help/api). This means you can use some of the Mastodon and Twitter clients for Friendica. The available features are client specific and may differ.
```
Did we remove the GNU Social extension from the Twitter like API that is supported by Friendica?

foss- replied:
https://github.com/friendica/friendica/pull/11010
